library/common/types/c_types.h

```diff
@@ -29,6 +29,11 @@ typedef enum { ENVOY_SUCCESS, ENVOY_FAILURE } envoy_status_t;
  */
 typedef enum { ENVOY_UNDEFINED_ERROR, ENVOY_STREAM_RESET } envoy_error_code_t;
 
+/**
+ * Networks by interface.
+ */
+typedef enum { ENVOY_WLAN, ENVOY_WWAN } envoy_network_t;
```

Comment:
Can we add docstrings for these? Also, I don't have strong thoughts here, but do you think `wifi` and `cellular` would be more readable?

Reply:
We debated about this for a while yesterday. I tend to agree that wifi and cellular are more readable, and easily understood by the reader. However, @goaway's point was that these actually represent correctness, albeit a bit pedantically. I am open to switching these to wifi and cellular.

Reply:
I guess I lean a bit towards WLAN and WWAN.

Some considerations that push me in that direction:
- WLAN and WWAN is consistent and correct (technically WiFi networks are cellular local networks)
- This is a purely internal struct and users of our library will never see it
- Apple uses WWAN in their enum (but confusingly also uses WiFi; though WiFi is the only widely-adopted general WLAN technology, newer IoT standards probably can be classified as WLAN as well)
- If we tune connection configuration to the technology, WWAN and WLAN are probably the classes that we care about

But it's also not that big a deal. I do agree we should at least provide a docstring here explaining that WLAN is, generally-speaking, WiFi and that mobile phone networks are all WWAN.

Reply:
This is fine with me, I don't feel that strongly 😃